astropy/utils/decorators.py

```diff
@@ -751,3 +752,230 @@ def _get_function_args(func, exclude_args=()):
                 all_args[arg_type] = None
 
     return all_args
+
+
+def format_doc(docstring, *args, **kwargs):
+    """
+    Replaces the docstring of the decorated object and then formats it.
+
+    This decorator keeps the objects signature and only alters the saved
+    ``__doc__`` property (so it cannot be used as class decorator in python2).
```

Comment:
Actually, this whole sentence is confusing, now that I re-read it.  Why would this decorator be expected to change signature in the first place?  I'd just remove it.